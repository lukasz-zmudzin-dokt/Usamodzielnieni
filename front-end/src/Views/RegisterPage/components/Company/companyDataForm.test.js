import {fireEvent, render} from "@testing-library/react";
import {MemoryRouter} from "react-router-dom";
import CompanyDataForm from "./companyDataForm";
import {getByTestId, getByText, waitForElement} from "@testing-library/dom";
import React from "react";


describe('CompanyDataForm', () => {
    it(' should render correctly', () => {
        const data = null;
        const onBlur = jest.fn();
        const { container } = render(
            <MemoryRouter>
                <CompanyDataForm data={data} onBlur={onBlur} />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });

    it('should return missing fields error', async () => {
        const data = parent.state.companyData;
        const onBlur = jest.fn();
        const {container, getByPlaceholderText} = render(
            <MemoryRouter>
                <CompanyDataForm data={data} onBlur={onBlur} />
            </MemoryRouter>
        );

        fireEvent.change(getByPlaceholderText("Nazwa firmy"), {
            target: {value: "qwe"}
        });
        fireEvent.change(getByPlaceholderText("Ulica"), {
            target: {value: "qwe"}
        });
        fireEvent.change(getByPlaceholderText("Nazwa miasta"), {
            target: {value: "qwe"}
        });
        fireEvent.change(getByPlaceholderText("Kod pocztowy"), {
            target: {value: "00-001"}
        });
        fireEvent.change(getByPlaceholderText("NIP"), {
            target: {value: "123"}
        });

        fireEvent.click(getByTestId(parent, "submitBtn"));
        await waitForElement(() => getByText(container, "Podaj "));
        expect(getByText("Podaj ", {exact: false})).toBeInTheDocument();
    });
});