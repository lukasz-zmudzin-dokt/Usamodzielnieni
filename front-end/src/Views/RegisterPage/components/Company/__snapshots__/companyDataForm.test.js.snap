// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDataForm should render correctly 1`] = `
<div>
  <div
    class="loginPage__cardSection px-0 col-lg mr-lg-3 card bg-light"
  >
    <h4
      class="card-header"
    >
      Dane firmy
    </h4>
    <div
      class="card-body"
    >
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupNameOfPlace"
          maxlength="60"
          minlength="1"
          name="name_of_place"
          placeholder="Nazwa firmy"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj nazwę firmy
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupStreet"
          maxlength="120"
          minlength="1"
          name="street"
          placeholder="Ulica"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj ulicę na której znajduje się firma
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupNumbah"
          maxlength="20"
          minlength="1"
          name="number"
          pattern="^([0-9]{1})[\\\\s\\\\S]*"
          placeholder="Numer budynku"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj numer budynku, przy którym znajduje się firma
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupCity"
          maxlength="40"
          minlength="1"
          name="city"
          placeholder="Nazwa miasta"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj nazwę miasta
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupCityCode"
          maxlength="6"
          minlength="1"
          name="city_code"
          pattern="[0-9]{2}[-][0-9]{3}"
          placeholder="Kod pocztowy"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj kod pocztowy w formacie 00-000
        </div>
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          id="formGroupCompanyNip"
          maxlength="10"
          minlength="10"
          name="company_nip"
          pattern="[0-9]{9}9"
          placeholder="NIP"
          required=""
          type="text"
        />
        <div
          class="invalid-feedback"
        >
          Podaj prawidłowy NIP
        </div>
      </div>
    </div>
  </div>
</div>
`;
